root/dataframe/test_trainCache.C

```diff
@@ -98,7 +98,57 @@ int WithTTreeReader()
    chain.GetTree()->PrintCacheStats();
 
    if (refNBytes != rbRAII.GetBytesRead()) {
-      std::cout << "With TChain: the bytes read (" << rbRAII.GetBytesRead()
+      std::cout << "With WithTTreeReader: the bytes read (" << rbRAII.GetBytesRead()
+                << ") are different from the reference (" << refNBytes << ")\n";
+      ret +=1;
+   }
+
+   return ret;
+}
+
+int WithTTreeReaderAndTwoBranches()
+{
+   std::cout << "\n----------- With TTreeReaderAndTwoBranches" << std::endl;
+
+   TChain chain(treeName);
```

Comment:
I would recommend that you use a TTree rather than a TChain for this example to avoid the doubt on whether the TTreeCache object is being reused between the two runs (as opposed to the TChain deleting and recreating it implicitly).